submissions/Nik3264/Building-a-Tiny-JS-World/index.js

```diff
@@ -0,0 +1,79 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _https://github.com/Nik3264/a-tiny-JS-world_
+   Web app: _https://nik3264.github.io/a-tiny-JS-world/_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const dog = {
+   species: 'dog',
+   name: 'Toby',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'woof-woof!'
+ };
+
+ const cat = {
+   species: 'cat',
+   name: 'Vasya',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'mi-mi!'
+ };
+ const catwoman = {
+   species: 'catwoman',
+   name: 'Toby',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: cat.saying
```

Comment:
Task asks you to make it so Catwoman always says the same thing the cat says. But in your current implementation, would Catwoman's saying change if we change the cat's one?

Reply:
Catwoman.saying’s  [[HomeObject]]  is cat because it was created in cat. The [[HomeObject]] property never changes. And when cat.saying changes -  catwoman.saying also changes, isn’t it? And why in this case it is needed to write __proto__: cat, ?  It is working now without writing it. When I change cat.saying, then catwoman.saying also changes immidiately. Catwoman.saying and cat.saying are interdependent because they are same concepts. We have the result - it works! Is there any issue? I consider it is but I don't know where.  :)  I have alredy read  some info about class inheritance, should I imply this concept here? Could you please give me some hint ? :)  Thank you a lot.

Reply:
What I mean is this:

```js
const cat = { saying: "First test" };
const catwoman = { saying: cat.saying };
console.log(catwoman.saying) // this will be "First test"
cat.saying = "Second test";
console.log(catwoman.saying) // this should be "Second test", but with your implementation, it will still be "First test"
```

You can use a variety of methods here, prototyping, `bind` etc. Just don't use classes yet.

In this particular task, you are right, it works like this, but the idea is to understand that simple using `saying: cat.saying` just gets the value of `cat.saying` and assigns it once, it is not a link and it will not be updated. And the task asks to make it so it does.

Reply:
Thank you! Now everything is clear. If the property is removed completely, then it is searched for in the prototype. So that's why it was necessary to manually create an array of properties! When using "Object.keys()"  the property "saying" there was none at all. Now I can clearly see the difference.